keras/src/backend/common/remat_scope.py

```diff
@@ -0,0 +1,127 @@
+from keras.src.backend.common import global_state
+
+
+class RematScope:
+    """A context manager for enabling rematerialization in Keras.
+
+    Rematerialization (gradient checkpointing) trades memory for computation by
+    recomputing intermediate activations during the backward pass. This is
+    particularly useful for training large models or large batch sizes within
+    limited memory constraints.
+
+    Args:
+        mode: Rematerialization mode to apply.
+            Options:
+            - "full": Apply rematerialization globally to all supported
+              operations.
+            - "activations": Apply rematerialization only to activation layers.
```

Comment:
Bit ambiguous, does it work with activations on another layer, e.g. `Dense`?

Reply:
yes, if a layer has keras.Activations specified, it is applied to it - https://github.com/keras-team/keras/blob/1073a2ead0857c54deded53e457d2656ac0205a4/keras/src/layers/layer.py#L1648

Reply:
Got it, please update the docstring to clarify this behavior